src/main/java/com/instana/opentracing/InstanaSpanContext.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+package com.instana.opentracing;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.opentracing.SpanContext;
+import io.opentracing.propagation.TextMapExtract;
+
+class InstanaSpanContext implements SpanContext {
+
+  private final Map<String, String> items;
```

Comment:
One might question the use of a `Map`, since it oversimplifies things. In general, HTTP header keys may appear multiple times. Using a `Map` leads to a last-seen-header-wins behaviour. I'd argue this is acceptable for the purpose at hand. Instana headers should only appear once. Other headers get filtered. 

In principle, it's sloppy to use a general `Map` for everything instead of doing a proper extraction of the information and storing it into separate fields. But I didn't want to stray too far from the original implementation.

To me, it's unclear if OpenTracing Baggage items themselves (while not being fully supported by Instana) have the semantics of a `Map`. If so, using a `Map` would actually be correct. I couldn't find any authoritative information on the topic.